atproto/lexicon/resolving_catalog.go

```diff
@@ -0,0 +1,74 @@
+package lexicon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/bluesky-social/indigo/atproto/identity"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+// Catalog which supplements an in-memory BaseCatalog with live resolution from the network
+type ResolvingCatalog struct {
+	Base      BaseCatalog
+	Directory identity.Directory
+}
+
+func NewResolvingCatalog() ResolvingCatalog {
+	return ResolvingCatalog{
+		Base:      NewBaseCatalog(),
+		Directory: identity.DefaultDirectory(),
+	}
+}
+
+func (rc *ResolvingCatalog) Resolve(ref string) (*Schema, error) {
```

Comment:
Can this string be anything? Should it be a URL or something and should we validate that?

Reply:
it is a "Lexicon reference", which is an NSID optionally followed by name-fragment. I'm planning to add a syntax parser for that identifier, but still hashing out the specification of the syntax (what we have written on atproto.com is probably too restrictive).